### PR TITLE
Enable to configure S3 client without environment variable

### PR DESCRIPTION
**Pull request by kirikiriyamama**
Hi, I would like to configure S3 client by using Rails configuration file (e.g. `config/secrets.yml`). It will be supported by this PR.

### CODE REVIEW

**Comment by ksss** [comment]
I'm not sure this reasons.

1. Break compatibility since `client` public method has been deleted.
1. You can use secrets data like this (I am also doing so).

in initializers/*.rb

```ruby
Aws.config[:region] = Rails.application.secrets.aws_region
```

**Comment by kirikiriyamama** [comment]
Does this line mean that environment variables are taken precedence over credentials file and instance profile?

> ENV["TORI_ACCESS_KEY"] > aws-sdk credentials

**Comment by ksss** [comment]
Yes.
Env takes precedence.

Ah..., Do you want to use access_key_id and secret_access_key without env ?

日本語でいいから問題を教えてw

**Comment by kirikiriyamama** [comment]
Oops, I didn't notice the first reason :bow: How about creating `attr_reader`?
About the second reason, I think that credential configurations should be become independent from global configurations. However, I don't have confidence in this opinion. Therefore, I'll close this PR and use the way which is proposed by you if you oppose strongly.

File: lib/tori/backend/s3.rb [hunk 1]
```diff
@@ -12,9 +12,27 @@ class S3
...
+        region =
+          options[:region] || ENV['TORI_AWS_REGION'] || ENV['AWS_REGION'] || Aws.config[:region]
+        @client =
+          case
+          when options[:access_key_id] && option[:secret_access_key]
+            Aws::S3::Client.new(
+              access_key_id:     options[:access_key_id],
+              secret_access_key: options[:secret_access_key],
+              region:            region)
+          when ENV['TORI_AWS_ACCESS_KEY_ID'] && options[:TORI_AWS_SECRET_ACCESS_KEY]
```
Comment (ksss) [inline lib/tori/backend/s3.rb]: `options[:TORI_AWS_SECRET_ACCESS_KEY]` looks not use.
This is `ENV['TORI_AWS_SECRET_ACCESS_KEY']` dont you ?

**Comment by ksss** [comment]
First is OK.
Second, I'm sure :access_key_id, :secret_access_key and :region options.

File: lib/tori/backend/s3.rb (hunk 1 above)
Comment (kirikiriyamama) [inline lib/tori/backend/s3.rb]: I'v fixed it :bow:

**Comment by kirikiriyamama** [comment]
Thanks :sparkles: I applied reviews.

File: lib/tori/backend/s3.rb [hunk 2]
```diff
@@ -6,15 +6,35 @@ module Backend
...
+        region =
+          options[:region] || ENV['TORI_AWS_REGION'] || ENV['AWS_REGION'] || Aws.config[:region]
+        @client =
+          case
+          when options[:access_key_id] && option[:secret_access_key]
+            Aws::S3::Client.new(
+              access_key_id:     options[:access_key_id],
+              secret_access_key: options[:secret_access_key],
+              region:            region)
+          when ENV['TORI_AWS_ACCESS_KEY_ID'] && ENV[:TORI_AWS_SECRET_ACCESS_KEY]
```
Comment (ksss) [inline lib/tori/backend/s3.rb]: This is `ENV['TORI_AWS_SECRET_ACCESS_KEY']` dont you ?
This code raise an error `TypeError: no implicit conversion of Symbol into String`

Comment (kirikiriyamama) [inline lib/tori/backend/s3.rb]: :bow: :bow: :bow:

File: lib/tori/backend/s3.rb [hunk 3]
```diff
@@ -82,46 +102,32 @@ def head(filename = nil)
       end
 
       def public_url(filename)
-        "#{client.config.endpoint}/#{@bucket}/#{filename}"
+        "#{@client.config.endpoint}/#{@bucket}/#{filename}"
```
Comment (ksss) [inline lib/tori/backend/s3.rb]: I think, This fix is that there is no relationship with this PR.
And other same change.

```diff
- client
+ @client
```

Could you split other PR or remove ?

Comment (kirikiriyamama) [inline lib/tori/backend/s3.rb]: Oh, these changes became unnecessary because attr_reader was added. I'll remove.

**Comment by ksss** [comment]
OK, Test was pass.
Thank you for contributing.

**Pull request merged**